Fix rendering of void elements

hibachrach opened the pull request:
Fixes #1. Note that this doesn't embed the fact that children of void elements are discarded/unused into the types--I figured that the friction for a developer of having to remember which elements are void feels a bit...trivia-esque. Open to other thoughts though!

hibachrach committed:
Fix rendering of void elements

lpil commented on src/htmb.gleam:
```diff
@@ -25,18 +25,33 @@ import gleam/list
+
+  // Void elements do not have a closing part and cannot accept children
+  // See https://www.w3.org/TR/2011/WD-html-markup-20110113/syntax.html#syntax-elements
+  case list.contains(void_elements, tag) {
```
Let's use pattern matching for this rather than seeking through a list, it'll be a little bit faster, and rendering HTML will be in the hot path for request/response

```gleam
case tag {
  "area" | "base" ... -> ...
  _ -> ...
}
```

hibachrach replied:
Happy to! Curious what the root cause is in the performance difference? Is this not manifested as a linked list by the compiler?

lpil reviewed (commented):
Thank you. Could you change the render_page function to always use the HTML5 doctype please?

lpil commented:
Oh! CI is failing because it's using an outdated version of Gleam. Would you mind bumping that too in the github workflow yml? Thanks

hibachrach committed:
Change target to be (exclusively) HTML5

BREAKING CHANGE: The `render_page` function no longer takes a `doctype`
parameter

hibachrach committed:
Update Gleam version used in GitHub Actions

lpil reviewed (approved):
Perfect! Thank you!
